src/requester.ts

```diff
@@ -0,0 +1,89 @@
+import bsv from 'bsv';
+import { AuthHeader, setSignature } from './authentication';
+import { Logger } from './logger';
+import { ErrorNoAdminKey, ErrorResponse } from './errors';
+
+export class Requester {
+  private logger: Logger;
+
+  private adminKey?: bsv.HDPrivateKey;
+  private signingKey?: bsv.HDPrivateKey | bsv.PrivateKey;
+  private xPubString?: string;
+  private url: string;
+
+  static CreateSigningRequester(
+    logger: Logger,
+    url: string,
+    signingKey?: bsv.HDPrivateKey | bsv.PrivateKey,
+    adminKey?: bsv.HDPrivateKey,
+  ) {
+    const requester = new Requester(logger, url);
+    requester.signingKey = signingKey;
+    requester.adminKey = adminKey;
+    return requester;
+  }
+
+  static CreateXPubRequester(logger: Logger, url: string, xPubString: string) {
+    const requester = new Requester(logger, url);
+    requester.xPubString = xPubString;
+    return requester;
+  }
+
+  private constructor(logger: Logger, url: string) {
+    this.logger = logger;
+    this.url = url.endsWith('/') ? url : url + '/'; //make sure the url ends with a '/'
+  }
+
+  async AdminRequest(path: string, method: string = 'GET', payload: any = null): Promise<any> {
```

Comment:
Maybe we could introduce two client types: "normal" and "admin", which would extand the normal client.
Thanks to this in requester we would have only one "request" method, and checking adminKey would be performed on client creation

Reply:
I've been talking to @dorzepowski about that. As I remember we decided not to do that. 
IMHO we can discuss it once again and maybe do it in some next task.

Generally, I like the idea 😄

Reply:
What we agreed was that, we can do something like that, but rather in a separate task, to finish this part to enable other works on the client, like bsv.js replacement.